webapp/client/cypress/integration/filing.spec.js

```diff
@@ -0,0 +1,64 @@
+describe("Filing", () => {
+  beforeEach(() => {
+    cy.login();
+  });
+
+  context("success", () => {
+    beforeEach(() => {
+      cy.fixture("est_data").then((est_data) => {
+        cy.request("POST", "/testing/set_session_data/form_data", est_data);
+      });
+      cy.visit("/lotse/step/filing");
+    });
+
+    it("downloading pdf is possible", () => {
+      cy.window()
+        .document()
+        .then(function (doc) {
+          doc.addEventListener("click", () => {
+            setTimeout(function () {
+              doc.location.reload();
+            }, 1000);
+          });
+          // Downloads folder is automatically cleared
+          cy.get("a").contains("Übersicht speichern").click();
+          cy.readFile("cypress/downloads/AngabenSteuererklaerung.pdf").should(
+            "exist"
+          );
+        });
+    });
+
+    it("going to next step is possible", () => {
+      cy.get("a").contains("Weiter").click();
+      cy.url().should("include", "/lotse/step/ack");
+    });
+  });
+
+  context("failure", () => {
+    beforeEach(() => {
+      cy.fixture("est_data").then((est_data) => {
+        est_data["produce_validation_error"] = true;
```

Comment:
Yes the C3PO id (we could of course also change that to another Id number.

> therefore we get another error
What other error? If the error is because of the C3P0, we could also think about another validation error or changing the INVALID_IDNR from C3PO to something closer to an actual Idnr.
I would expect that the idnr send to MockErica would be independent from the one, that is used for login.

Imo, we should either explicitly set the behavior of the MockEric through a flag (which is so far not possible to do for the functional tests) or try to use input data that would also lead to some problem in the actual Erica

Reply:
But we are using the login fixture because otherwise we couldn't access the StepFiling page. And we are logging in with a valid tax number: `"idnr": ["04", "452", "397", "687"]`. If we then use the C3PO fake id to post the testing data we get the error:
`Bitte prüfen Sie die Steuer-Identifikationsnummer. Die Angabe muss mit der Nummer identisch sein, die Sie zum Anmelden eingegeben haben.`
So to my understanding we need to use the same id for both logging in and posting the testing data right?

Reply:
Discussed in the tech daily. Producing an error would be leant to a real case and added in the mock erica